mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/model/impl/AbstractPojoModelCompositeElement.java

```diff
@@ -44,7 +48,7 @@
 	@SuppressWarnings("unchecked") // The cast is checked using reflection
 	public final <T> PojoModelElementAccessor<T> createAccessor(Class<T> requestedType) {
 		if ( !isAssignableTo( requestedType ) ) {
-			throw new SearchException( "Requested incompatible type for '" + createAccessor() + "': '" + requestedType + "'" );
+			log.incompatibleRequestedType( createAccessor(), requestedType );
```

Comment:
Could you please improve this while you're at it; I would expect to pass a path here, not the accessor itself.

Maybe do something like this:

```
log.incompatibleRequestedType( getModelPathTypeNode().toUnboundPath(), requestedType );
```

... and use `PojoModelPathFormatter` to format the path.